src/pages/blog/post/der-komplette-guide-rund-um-sql-injections.mdx

```diff
@@ -0,0 +1,303 @@
+* **2013**: [OWASP Top 10: Platz #1 (A1 – Injection)](https://owasp.org/www-pdf-archive//OWASP_Top_10_-_2013.pdf)
+* **2017**: [OWASP Top 10: Platz #1 (A1 - Injection)](https://owasp.org/www-project-top-ten/2017/Top_10)
+* **2021**: [OWASP Top 10: Platz #3 (A3 - Injection)](https://owasp.org/www-project-top-ten/)
+
+Weiterhin wurden zum aktuellen Stand (19. Februar 2023), 1789 Sicherheitslücken, die auf SQL-Injections zurückzuführen sind, in der [offiziellen CVE-Datenbank allein für das Jahr 2022 gemeldet](https://www.cvedetails.com/vulnerability-list/year-2022/opsqli-1/sql-injection.html).
+
+## Was sind die Gründe, warum diese Sicherheitslücke noch nicht ausgestorben ist?
+
+1. **Fehlende Wissen**: Vielen Software-Entwicklern fehlt oft das benötigte Know-How und eine gewisse Erfahrung und im Bereich Sicherheit und Sicherheitsbewusstsein
+
+2. **Legacy Code**: Viele Webanwendungen beruhen immer noch auf altem Code, der entwickelt wurde, bevor Best-Practices zur Absicherung eingeführt wurden. Entsprechende Eingabevalidierung oder andere Sicherheitsmaßnahmen, was ihn anfällig für SQL-Injection-Angriffe macht, fehlt dort.
+
+3. **Gesteigerte Komplexität**: Webanwendungen sind im Laufe der Zeit immer komplexer geworden, mit mehr Komponenten und Abstraktionsebenen. Schwachstellen zu erkennen und zu entschärfen, auch solche, die mit SQL-Injection zusammenhängen, ist dadurch deutlich erschwert.
+
+4. **Automatisiertes Testen**: Automatisierte und effiziente Testmethoden zur Erkennung von SQL-Injections (z.B. ohne False Positives) sind kaum vorhanden bzw. nicht verbreitet.
+
+5. **API-Design**: Client-Libraries, die zum Verbindungsaufbau zur Datenbank genutzt werden, können oft immer noch missbraucht oder falsch angewendet werden und geben (unter Umständen) dem Entwickler ein falsches Gefühl von Sicherheit.
+
+6. **Menschliches Versagen**: Entwickler sind auch nur Menschen und Bug-Freie Software zu schreiben ist auch noch nicht 100% möglich. So kann ein Entwickler beispielsweise vergessen, Benutzereingaben zu bereinigen, oder ein Sicherheitsteam räumt der Behebung von Sicherheitslücken nicht rechtzeitig Priorität ein.
+
+7. **Attraktives Angriffsziel und hartnäckige Angreifer**: Fast jede Webanwendung verwendet irgendeine Form von Datenbank im Backend, was es somit zu einem lohnenswerten Angriffsziel macht. Zusätzlich können Angreifer eine Vielzahl von Techniken, einschließlich automatisierter Tools, verwenden, um SQL-Injection-Schwachstellen zu erkennen und auszunutzen.
+
+## Welcher Schaden kann durch eine SQL-Injection angerichtet werden?
+
+Der genaue Schaden bzw. die Möglichkeiten hängen von der entsprechenden Umgebung (z. B. der Datenbank selbst), dem Ausmaß der entsprechenden Lücke sowie den existierenden Sicherheitsvorkehrungen ab.
+
+Dies kann zu folgenden Szenarien führen:
+
+* Datendiebstahl: Sensible Daten wie Anmeldedaten, Kreditkartennummern und persönliche Informationen
```

Comment:
...Informationen gelangen in die falschen Hände

Reply:
```suggestion
* Datendiebstahl: Sensible Daten wie Anmeldedaten, Kreditkartennummern und persönliche Informationen gelangen in die falschen Hände
```